src/main/java/net/dv8tion/jda/core/handle/PresenceUpdateHandler.java

```diff
@@ -90,31 +102,29 @@ protected Long handleInternally(JSONObject content)
             String gameName = null;
             String gameUrl = null;
             Game.GameType type = null;
-            if ( !content.isNull("game") && !content.getJSONObject("game").isNull("name") )
+            final JSONObject game = content.optJSONObject("game");
+            if (game != null && !game.isNull("name"))
             {
-                gameName = content.getJSONObject("game").get("name").toString();
-                gameUrl = ( content.getJSONObject("game").isNull("url") ? null : content.getJSONObject("game").get("url").toString() );
+                gameName = game.get("name").toString();
+                gameUrl = game.isNull("url") ? null : game.get("url").toString();
                 try
                 {
-                    type = content.getJSONObject("game").isNull("type")
+                    type = game.isNull("type")
                             ? Game.GameType.DEFAULT
-                            : Game.GameType.fromKey(Integer.parseInt(content.getJSONObject("game").get("type").toString()));
+                            : Game.GameType.fromKey(Integer.parseInt(game.get("type").toString()));
```

Comment:
Wouldn't `game.getInt("type")` be the same here?

Reply:
In the past we've encountered numerous issues with game objects due to them not being checked by the API, I didn't touch that logic at all here.